pkg/trace/writer/trace_test.go

```diff
@@ -42,38 +43,45 @@ func (s MockSampler) GetTargetTPS() float64 {
 var mockSampler = MockSampler{TargetTPS: 5, Enabled: true}
 
 func TestTraceWriter(t *testing.T) {
-	srv := newTestServer()
-	cfg := &config.AgentConfig{
-		Hostname:   testHostname,
-		DefaultEnv: testEnv,
-		Endpoints: []*config.Endpoint{{
-			APIKey: "123",
-			Host:   srv.URL,
-		}},
-		TraceWriter: &config.WriterConfig{ConnectionLimit: 200, QueueSize: 40},
-	}
+	testCases := []bool{false, true}
 
-	t.Run("ok", func(t *testing.T) {
-		testSpans := []*SampledChunks{
-			randomSampledSpans(20, 8),
-			randomSampledSpans(10, 0),
-			randomSampledSpans(40, 5),
+	for _, tc := range testCases {
+		srv := newTestServer()
+		cfg := &config.AgentConfig{
+			Hostname:   testHostname,
+			DefaultEnv: testEnv,
+			Endpoints: []*config.Endpoint{{
+				APIKey: "123",
+				Host:   srv.URL,
+			}},
+			TraceWriter: &config.WriterConfig{ConnectionLimit: 200, QueueSize: 40},
 		}
-		// Use a flush threshold that allows the first two entries to not overflow,
-		// but overflow on the third.
-		defer useFlushThreshold(testSpans[0].Size + testSpans[1].Size + 10)()
-		tw := NewTraceWriter(cfg, mockSampler, mockSampler, mockSampler, telemetry.NewNoopCollector(), &statsd.NoOpClient{}, &timing.NoopReporter{})
-		tw.In = make(chan *SampledChunks)
-		go tw.Run()
-		for _, ss := range testSpans {
-			tw.In <- ss
+		if tc {
+			cfg.Features = map[string]struct{}{"zstd-encoding": {}}
 		}
-		tw.Stop()
-		// One payload flushes due to overflowing the threshold, and the second one
-		// because of stop.
-		assert.Equal(t, 2, srv.Accepted())
-		payloadsContain(t, srv.Payloads(), testSpans)
-	})
+
+		t.Run("ok", func(t *testing.T) {
```

Comment:
Maybe something like `t.Run(fmt.Sprintf("zstd_%t", tc), func(t *testing.T) {` to differentiate the test cases in the logs.

Reply:
Fixed thanks a lot.